### DISCUSSION TITLE
[nuxt3 rc4] How to include .env.production on local build?

### DISCUSSION THREAD

**Discussion by dennisbaum** [nuxt/nuxt#16402] [Questions]
Can we define .env.prouction, to be used on a local build by nuxi build?
rather than renaming .env.production to .env...
thx!

**Comment by Flanker72** [comment]
Seems it should work out of the box. Nuxt is based on Vite bundler, and Vite has the support of env modes: https://vitejs.dev/guide/env-and-mode.html#env-files

**Comment by dennisbaum** [comment]
i thought so too, but unfortunately there is no change, when i use nuxi build. (which sets NODE_ENV=production automatically) 
<img width="269" alt="grafik" src="https://user-images.githubusercontent.com/562826/177360739-41988c84-30a5-4083-bd8d-74a1ab54d322.png">

```
runtimeConfig: {
    public: {
      baseUrl: '',
      testUrl: process.env.TEST_URL,
    },
  },
```

```js
const runtimeConfig = useRuntimeConfig()
console.log(runtimeConfig.public.testUrl, process.env.NODE_ENV);
```
runtimeConfig still logs the variables from `.env` when i preview by nuxi preview

it also does not take effect, if i use something like...
```shell
NODE_ENV=production nuxi build
NODE_ENV=production nuxi preview
nuxi build --mode production
```

**Reply by Flanker72** [reply]
Seems it's some kind of a bug. Let's wait for developers' reaction. I think, you may try to create an issue about such a behavior.

**Comment by Der-Alex** [comment]
As far as I understood from the docs: https://v3.nuxtjs.org/guide/features/runtime-config#environment-variables

You have to prefix your env variables like `NUXT_{{VAR}}` or `NUXT_PUBLIC_{{VAR}}` for env vars that should be exposed to the client side.

But I cannot get it to work. I have a .env.development file but nuxt only loads data from .env when using `useRuntimeConfig()`

**Comment by BulgarianHealer** [comment]
Still has this 'bug/problem', tested with rc13.

**Reply by goshku** [reply]
> Still has this 'bug/problem', tested with rc13.

Same here :)

**Reply by TheAlexLichter** [reply]
Please open an issue then with reproduction ☺️
